### ISSUE TITLE
ERROR

### ISSUE THREAD

**Issue by 1661145410** [KIT-IBT/vtkToolbox#3]
when i finished the install of toolbox with cmake, i run the /toolbox/matlab/vtk_mex_build.m to install mex. but in matlab there was an error:  

Wrong use of vertcat

The dimensions of the array to be concatenated are inconsistent.

Error vtk_ mex_ Build (line 47)

flags = { ...

i have checked but no clue.
is anyone can help?

**Comment by MichaelLoibl** [comment]
I am also currently trying to get it run.

There is "..." missing in one of the lines between line 47 and 95.

Furthermore, I changed `['-L' vtkBinPath '/lib']` to `['-l' vtkBinPath '/lib']`.

But resolving this I got stuck again with the error:

Error using mex
$$LIB not detected; check that you are in the correct current folder, and check the spelling of '$$LIB'.

Error in vtk_mex_build (line 99)
mex('../vtkRead/vtkRead.cxx', '../libvtkMatlab/vtkToStruct.cxx', '../libvtkMatlab/common.cxx', flags{:}); % '../vtkRead/additionalReaders/vtkOFFReader.cxx'

**Comment by axel-loewe** [comment]
> Error vtk_ mex_ Build (line 47)
> 
> flags = { ...

Line 47 is commented in the default code (assuming installation of VTK from Macports). If you install it from source, you should comment the section assuming installation from Macports. Does that help?

**Comment by axel-loewe** [comment]
> There is "..." missing in one of the lines between line 47 and 95.
> 
Thanks, fixed now. We also welcome pull requests for changes.

> Furthermore, I changed `['-L' vtkBinPath '/lib']` to `['-l' vtkBinPath '/lib']`.
> 
Are you sure this is what you want to do (providing directories for the linker vs. telling which libraries to link). I feel that this might related to the error you are seeing

**Comment by MichaelLoibl** [comment]
That's right. My fault. I already changed it back and got it successfully compiled.

However, I have still problems to use it. I get the following error (in German which you might understand :) ):
"Invalid MEX-file 'C:\Users\Loibl\AppData\Roaming\MathWorks\MATLAB Add-Ons\Collections\vtkToolbox\MATLAB\vtkRead.mexw64': Das
angegebene Modul wurde nicht gefunden."

I try to call the respective file from a script which I put in the same folder together with a .vtk-file. Do you add certain libraries to the system path?

**Comment by axel-loewe** [comment]
Sorry, we don't have experience with Windows deployment for this tool
